### PR TITLE
[REVIEW] fix gcc-9 compilation errors on tests

### PR DESCRIPTION
**Pull request by rongou**
@harrism @jrhemstad @kkraus14

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: cpp/tests/io/orc_test.cpp [hunk 1]
```diff
@@ -703,20 +703,20 @@ TYPED_TEST(OrcChunkedWriterNumericTypeTest, UnalignedSize)
   bool mask[] = {0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
                  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1};
 
-  T c1a[] = {5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5,
-             5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5};
-  T c1b[] = {6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6,
-             6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6};
+  T c1a[num_els];
+  for (auto& e : c1a) { e = static_cast<T>(5); }
```
Comment (karthikeyann) [inline cpp/tests/io/orc_test.cpp:707]: suggesting std::generate

File: cpp/tests/copying/copy_tests.cu [hunk 2]
```diff
@@ -299,11 +299,11 @@ TYPED_TEST(CopyTestNumeric, CopyIfElseTestScalarColumn)
...
   cudf::numeric_scalar<T> lhs_w(5);
 
-  T rhs[]      = {6, 6, 6, 6};
+  T rhs[]      = {static_cast<T>(6), static_cast<T>(6), static_cast<T>(6), static_cast<T>(6)};
   bool rhs_v[] = {1, 0, 1, 1};
   wrapper<T> rhs_w(rhs, rhs + num_els, rhs_v);
 
-  T expected[] = {5, 6, 6, 5};
+  T expected[] = {static_cast<T>(5), static_cast<T>(6), static_cast<T>(6), static_cast<T>(5)};
   wrapper<T> expected_w(expected, expected + num_els, rhs_v);
```
Comment (karthikeyann) [inline cpp/tests/copying/copy_tests.cu:307]:
```suggestion
const auto expected = cudf::test::make_type_param_vector<TypeParam>({5, 6, 6, 5});
  wrapper<T> expected_w(expected.begin(), expected.end(), rhs_v);
```

File: cpp/tests/copying/copy_tests.cu [hunk 3]
```diff
@@ -299,11 +299,11 @@ TYPED_TEST(CopyTestNumeric, CopyIfElseTestScalarColumn)
 
   cudf::numeric_scalar<T> lhs_w(5);
 
-  T rhs[]      = {6, 6, 6, 6};
+  T rhs[]      = {static_cast<T>(6), static_cast<T>(6), static_cast<T>(6), static_cast<T>(6)};
```
Comment (karthikeyann) [inline cpp/tests/copying/copy_tests.cu:302]:
```suggestion
const auto rhs = cudf::test::make_type_param_vector<TypeParam>({6, 6, 6, 6});
```

Comment (karthikeyann) [reply cpp/tests/copying/copy_tests.cu:302]: similar suggestions where multiple `static_cast<T>(` is present.

File: cpp/tests/io/parquet_test.cpp [hunk 4]
```diff
@@ -873,20 +873,20 @@ TYPED_TEST(ParquetChunkedWriterNumericTypeTest, UnalignedSize)
   bool mask[] = {0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
                  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1};
 
-  T c1a[] = {5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5,
-             5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5};
-  T c1b[] = {6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6,
-             6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6};
+  T c1a[num_els];
+  for (auto& e : c1a) { e = static_cast<T>(5); }
```
Comment (karthikeyann) [inline cpp/tests/io/parquet_test.cpp:877]: std::generate

**Review by karthikeyann — CHANGES_REQUESTED** [review]

File: cpp/tests/copying/copy_tests.cu (hunk 3 above)
Comment (rongou) [reply cpp/tests/copying/copy_tests.cu]: Done.

File: cpp/tests/copying/copy_tests.cu (hunk 2 above)
Comment (rongou) [reply cpp/tests/copying/copy_tests.cu]: Done.

File: cpp/tests/io/orc_test.cpp (hunk 1 above)
Comment (rongou) [reply cpp/tests/io/orc_test.cpp]: Switched to `std::fill`

File: cpp/tests/io/parquet_test.cpp (hunk 4 above)
Comment (rongou) [reply cpp/tests/io/parquet_test.cpp]: ditto

**Comment by rongou** [comment]
rerun tests

**Review by karthikeyann — APPROVED** [review]

**Review by harrism — APPROVED** [review]
Nice cleanup!

**Pull request merged**